cfpb/cfgov-refresh
# Create BrowseFilterablePage and FilterForm

- Add BrowseFilterablePage model
- Add FilterForm form for filterable list
- Add BaseExpandable class for expandable controls
- Add FilterControls organism using BaseExpandable
- Add url_parameters macro to handle adding existing get URL parameters into links
- Change jinja2 templates to handle Wagtail page
 - browse-filterable.html
 - filterable-list-controls.html
 - post-preview.html
 - notification.html
 - pagination.html

## TEST
Test 1:
- Create a BrowseFilterablePage
- Create Learn, Doc Detail, or Event pages as descendants (child pages) of the BrowseFilterablePage
- Play around with the controls to filter the list

Test 2:
- Create Create Learn, Doc Detail pages with Blog type categories (not as descendants) of BrowsefilterablePage
- Select `Newsroom` as the page type in the filter controls
- Go to filter page to check if they are there (they should be)
 - Additionally check that they have the Blog category (not their specific one)

Test 3
- Test Preview functionality works

Test 4
- Make sure that sharing works by only "sharing" the pages then going to the filter page at the `content.localhost` site and finding them. They should show as well, but not if going to the `localhost` site.


@KimberlyMunoz 
@sebworks 
@jimmynotjim 
@anselmbradford 
@kave 
@rosskarchner
